### PR TITLE
Add linear api in docs

### PR DESCRIPTION
**Pull request by tmigot**
Modify the doc following #379 #377

### CODE REVIEW

**Review by abelsiqueira — CHANGES_REQUESTED** [review]
Thanks for the quick PR.

File: docs/src/guidelines.md [hunk 1]
```diff
@@ -64,11 +64,11 @@ The following functions should be defined:
...
-  - `cons!(nlp, x, c)`
-  - `jac_structure!(nlp, jrows, jcols)`
-  - `jac_coord!(nlp, x, jvals)`
-  - `jprod!(nlp, x, v, Jv)`
-  - `jtprod!(nlp, x, v, Jtv)`
+  - `cons_nln!(nlp, x, c)`
+  - `jac_nln_structure!(nlp, jrows, jcols)`
+  - `jac_nln_coord!(nlp, x, jvals)`
+  - `jprod_nln!(nlp, x, v, Jv)`
+  - `jtprod_nln!(nlp, x, v, Jtv)`
```
[the remark this answers is not in the document]
Comment (tmigot) [reply docs/src/guidelines.md:71]: In theory no, for instance [L.57](https://github.com/JuliaSmoothOptimizers/NLPModels.jl/blob/289b81ceea0961cb60a6e536d32311114b09405b/src/nlp/api.jl#L57), it doesn't call `cons_lin` (resp. `cons_nln`) if you have no linear (resp. nonlinear) constraints.

[the remark this answers is not in the document]
Comment (abelsiqueira) [reply docs/src/guidelines.md:71]: That's the second situation in my list, you have to make sure that `nlp.meta.lin = []`. My first comment ("the default will error") is not totally accurate, but the additional information still applies.

[the remark this answers is not in the document]
Comment (abelsiqueira) [reply docs/src/guidelines.md:71]: Yes, I think something like "`cons_nln!(...)` and `cons_lin!(...)`", and then after the list is complete, we add disclaimers like "Notice that if you need to define the linear constraints by using `nlp.meta.lin`. If your model doesn't differentiate between linear and nonlinear constraints you can define only the `*_nln!` variants above, or alternatively, the functions without the suffixes `_nln!` (e.g., only `cons!`)."

[the remark this answers is not in the document]
Comment (tmigot) [reply docs/src/guidelines.md]: Ok, I tried a new version. Let me know what you think.

**Review by abelsiqueira — APPROVED** [review]
Looks good to me!

**Comment by abelsiqueira** [comment]
Thanks

**Pull request merged**